### PR TITLE
[10.0][ADD] stock_change_price_at_date

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

**Comment by rousseldenis** [comment]
@Cedric-Pigeon

File: stock_change_price_at_date/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            products_by_id = []
+            for result in results:
+                products_by_id.append(result.get('product_id')[0])
+            products_to_write = self.env['product.product'].browse()
+            for product in self:
+                if product.id not in products_by_id:
+                    products_to_write |= product
+            super(ProductProduct, products_to_write).write(vals)
+        vals.pop('standard_price')
+        return super(ProductProduct, self - products_to_write).write(vals)
```
Comment (Cedric-Pigeon) [inline stock_change_price_at_date/models/product_product.py]: what if `self - products_to_write` leads to an empty recordset ?

File: stock_change_price_at_date/tests/test_change_price_at_date.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        inventory.action_done()
+
+    def _change_price(self, new_price, move_date=None, template=False):
+        """
+        Launch wizard to change price
+        Set date in context for moves if filled in
+        :param new_price: New Price
+        :param move_date: The date of the account move
+        """
+        if template:
```
Comment (Cedric-Pigeon) [inline stock_change_price_at_date/tests/test_change_price_at_date.py:86]: I would protect this method by `self.ensure_one()`

File: stock_change_price_at_date/tests/test_change_price_at_date.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+                force_period_date=fields.Date.to_string(move_date),
+            )
+        self.wizard = wizard_context.create({
+            'counterpart_account_id': self.account.id,
+            'new_price': new_price,
+            'date': move_date or False})
+        self.wizard.change_price()
+
+    def _get_move_line_domain(self, template=False):
+        product_domain = [('product_id', '=', self.product.id)]
```
Comment (Cedric-Pigeon) [inline stock_change_price_at_date/tests/test_change_price_at_date.py:109]: Same remark hre, I would ensure a singleton

**Review by Cedric-Pigeon — CHANGES_REQUESTED** [review]
Small remarks

File: stock_change_price_at_date/tests/test_change_price_at_date.py (hunk 2 above)
Comment (rousseldenis) [reply stock_change_price_at_date/tests/test_change_price_at_date.py:86]: Test function

**Comment by rousseldenis** [comment]
@Cedric-Pigeon

**Review by Cedric-Pigeon — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
Tested on runbot

**Pull request merged**